### ISSUE TITLE
[Bug] API returns incorrect data type

### ISSUE THREAD

**Issue by 0xThiebaut** [cobbr/Covenant#163]
**Feature Request or Bug**
Bug

**Describe the feature request or bug**
When querying the `/api/users` endpoint, the returned data does not correspond to the Swagger definition as some missing fields are `null` instead of an empty string.

```bash
curl -X GET --header 'Accept: application/json' --header 'Authorization: Bearer eyJ[...]iGc' 'https://covenant:7443/api/users'
```

> Sample response:
> ```json
> [
>   {
>     "id": "c269f456-c5ba-45b8-b5e7-b3ed0cc4e571",
>     "userName": "student",
>     "normalizedUserName": "STUDENT",
>     "email": null,
>     "normalizedEmail": null,
>     "emailConfirmed": false,
>     "passwordHash": "",
>     "securityStamp": "NBGZZDVM4AL46PHE26DC3KDD4WB4GE7I",
>     "concurrencyStamp": "62597f14-0047-4ae9-8cdc-2a85005553d5",
>     "phoneNumber": null,
>     "phoneNumberConfirmed": false,
>     "twoFactorEnabled": false,
>     "lockoutEnd": null,
>     "lockoutEnabled": true,
>     "accessFailedCount": 0
>   }
> ]
> ```

The impact is that Swagger clients such as Python's [`bravado`](https://github.com/Yelp/bravado) can' t validate the returned data:

> jsonschema.exceptions.ValidationError: None is not of type 'string'
> 
> Failed validating 'type' in schema['items']['properties']['email']:
>     {'type': 'string'}
> 
> On instance[0]['email']:
>     None
> 

**To Reproduce**

With [`bravado`](https://github.com/Yelp/bravado) installed, the following Python 3 code to retrieve users fails.
```python
from urllib.parse import urlparse

from bravado.requests_client import RequestsClient
from bravado.client import SwaggerClient

api = 'https://covenant:7443/swagger/v1/swagger.json'


def main():
    http_client = RequestsClient()
    http_client.ssl_verify = False
    client = SwaggerClient.from_url(
        api,
        http_client=http_client,
    )
    login = client.CovenantUserApi.ApiUsersLoginPost(
        login={"userName": "student", "password": "secret"}).response().result
    bearer = login.covenantToken
    http_client.set_api_key(host=urlparse(api).netloc, api_key=f'Bearer {bearer}', param_name="Authorization",
                            param_in="header", )
    users = client.CovenantUserApi.ApiUsersGet().response().result
    print(users)


if __name__ == "__main__":
    main()

```

**Expected behavior**
I would expect missing fields to have a valid default value. As such, fields of the `string` type should have an empty string (`""`) instead of `null`. Furthermore, should the field have a format (i.e. the `lockoutEnd`), I would expect the response to have a similar format (i.e. default epoch time). Alternatively, one may leverage the OpenAPI optional/required fields which are used in the specification, although I couldn't find their own section.

```json
[
  {
    "id": "c269f456-c5ba-45b8-b5e7-b3ed0cc4e571",
    "userName": "student",
    "normalizedUserName": "STUDENT",
    "email": "",
    "normalizedEmail": "",
    "emailConfirmed": false,
    "passwordHash": "",
    "securityStamp": "NBGZZDVM4AL46PHE26DC3KDD4WB4GE7I",
    "concurrencyStamp": "62597f14-0047-4ae9-8cdc-2a85005553d5",
    "phoneNumber": "",
    "phoneNumberConfirmed": false,
    "twoFactorEnabled": false,
    "lockoutEnd": "1970-01-01T00:00:00Z",
    "lockoutEnabled": true,
    "accessFailedCount": 0
  }
]
```
**Covenant Server Information:**
 - OS: Ubuntu
 - Docker or Native: Self-compiled Docker image

**Comment by cobbr** [comment]
Thanks for the heads up @0xThiebaut . This one will take a bit of thinking, these fields are generated by AspNetCore, not in Covenant.

**Comment by cobbr** [comment]
I think this should be fixed in 8ba8b6e1343cff569f81e46cebae955b6355e2a5 in dev branch, will merge to master soon.

Can you verify that this fixes the issue for you @0xThiebaut?

**Comment by cobbr** [comment]
Going to close this for now, let me know if the issue still persists and we can re-open.
